pyiron_atomistics/vasp/base.py

```diff
@@ -2443,6 +2443,11 @@ def __init__(self, input_file_name=None, table_name="kpoints"):
         self._path_name = None
         self._n_path = None
 
+    def write_file(self, file_name, cwd=None):
+        if self.file_name == False:
```

Comment:
To maintain type consistency, I'd avoid assigning boolean values to `file_name` but rather set it to `None`

```suggestion
        if self.file_name is None:
```

Reply:
The problem is that None is the default value right now, so using that would always lead to the file not being written. I could also change the default to "KPOINTS" maybe.

Reply:
> It would be nice to write a simple unittest for jobs like this to check it it works [(some examples here)](https://github.com/pyiron/pyiron_atomistics/blob/master/tests/vasp/test_vasp.py). Additionally, we should warn users when 'KSPACING' is there in the INCAR file but a KPOINTS file is also present

Alternatively to what I did it would also be possible to only write the KPOINTS file if "KSPACING" is not present in the INCAR and issue a warning

Reply:
I would suggest to make a change around [here](https://github.com/pyiron/pyiron_atomistics/blob/907991b31ba4543ffb90a33d957108181e520506/pyiron_atomistics/vasp/base.py#L1859) and issue a warning if no KPOINTS is written.  In the current solution its not at all clear how this forces the file not being written and will probably be forgotten next week.